ompi/mca/pml/ob1/pml_ob1_recvreq.c

```diff
@@ -449,6 +454,7 @@ static void mca_pml_ob1_rget_completion (mca_btl_base_module_t* btl, struct mca_
 static int mca_pml_ob1_recv_request_put_frag (mca_pml_ob1_rdma_frag_t *frag)
 {
     mca_pml_ob1_recv_request_t *recvreq = (mca_pml_ob1_recv_request_t *) frag->rdma_req;
+    ompi_communicator_t *comm = recvreq->req_recv.req_base.req_comm;
```

Comment:
This will generate warning when MCA_BASE_EVENT are disabled.

Reply:
@bosilca Sorry, I'm having difficulty disabling MCA_BASE_EVENT.  Can you provide me with some guidance here?

Reply:
Well, I don't think it is possible to turn them off. Going back to look into the code I have a new set of questions:
1. How to you turn the events off ? I see we can turn them on/off with the callbacks, but there seem to be no way to completely turn them off. 
2. Why are they implemented in opam/mca/base/mca_base_event*. These are not MCA base type of things, what goes in the base is related to the MCA params and aliases, the events looks too low level for me. In fact why are they not at the MPI level ?

Reply:
Looking again at the code I don't see any way to turn these events off at compile time (such that we totally compile them off from). Which means they will always be in, and we have to pay the cost, even when they are not in use. 

But looking at the code also raised additional questions. Why are the events MCA base level ? Why are they even related to the MCA ? I don't think this is the right place to have them in, I don't even think they deserve to be at the OPAL level (because they are only used in OMPI).

Reply:
For 1 : I have implemented an --enable-mpi_t_events configure flag that disables implemented events in pml/ob1 and osc.  The MPI_T Events functions are still accessible, there just aren't any available events.  I'm hoping to make this available shortly.

For 2 : It looks to me like Nathan used the existing opal pvar code to implement MPI_T Events. My understanding is that events will be marked invalid if modules are not used.

Reply:
When your `--enable-mpi_t_events` will be turned off the comm variable you declare here will be unused, triggering a compiler complaint.

MCA stands for Modular Component Architecture, and I'm not sure how these MPI events could become part of that. We tried hard to separate things by scope, to facilitate the code readability, and from this perspective these MPI events do not belong into the MCA base. I'm sure others in the community have comments/suggestions.

Reply:
OK, sounds like this is going to need input from others, such as @jsquyres or @hppritcha.  I will hold off on my update until we address this.  @awlauria I'm also assuming that this will not be able to be implemented in time for 5.0.

Reply:
@jsquyres @hppritcha @bosilca Any comments or suggestions on moving forward with this?  The MPI_T Events backend functionality has been implemented in OPAL, similar to the MPI_T Performance variable implementation, to allow registration of events based on available modules.

Reply:
This is exactly my point, why an `MPI_*` level functionality exists in OPAL ? There is no portability issues there, and these events are extremely MPI-facing (anybody else would not care about calling a specialized callback allowing an upper layer to track events).

Reply:
I'm looking at seeing if there's a chance that we can rescue this PR.  From @bosilca comments it sounds like there are objections to the basic design for MPI_T events - using the mca variable registration infrastructure in opal.  It looks to me like the architect of most of the MPT_T pvar and now events decided to make use of the mca variable management mechanism since the inception of MPI_T support in OMPI.  Do we want to redesign this?  I'm noticing use of these pvar in both opal and ompi.